### PR TITLE
Merge Folder widget

### PR DESCRIPTION
**Pull request by stulees**
- Merged, revised logic, and consolidated or separated all single and folder code where applicable. No duplicate code, predominantly shared code base
- Merged, refactored, and improved upon unit tests
- Merged and revised integration tests
- No new version required, Widget handles `video.pause` missing from settings. Until settings are updated to include the "resume playing after" setting, it can not and does not get applied

### CODE REVIEW

**Comment by stulees** [comment]
@rodrigopavezi @donnapep Please review, thanks

File: src/widget/video.js [hunk 1]
```diff
@@ -199,41 +240,45 @@ RiseVision.Video = (function (gadgets) {
...
+        _storage = new RiseVision.Video.StorageFile(_additionalParams);
+        _storage.init();
       }
     }
+    else if (_mode === "folder") {
+      // create and initialize the Storage folder instance
+      _storage = new RiseVision.Video.StorageFolder(_additionalParams);
+      _storage.init();
+    }
+
```
Comment (rodrigopavezi) [inline src/widget/video.js]: @stulees curious, we don't support folders from a non rise storage? cheers

Comment (stulees) [inline src/widget/video.js]: @rodrigopavezi That's correct, we only support folders from a Storage account.

**Comment by rodrigopavezi** [comment]
@stulees looks good! :+1:  cheers

File: src/widget/video.js (hunk 1 above)
Comment (rodrigopavezi) [inline src/widget/video.js]: @stulees thanks for the clarification!!

**Comment by stulees** [comment]
@donnapep @rodrigopavezi Please review last commit, thanks

**Comment by donnapep** [comment]
:+1:

**Pull request merged**